utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameter.kt

```diff
@@ -0,0 +1,6 @@
+package org.oppia.android.util.platformparameter
+
+/** Generic interface whose implementations will be provided as Platform Parameter Values. */
```

Comment:
This might need some more detail to explain how they correspond to the defined protos and why we want this interface.

Reply:
Okay

Reply:
Hm maybe `Generic interface that is used to provide Platform Parameter Values corresponding to the [PlatformParameter] proto values. Objects that implement this interface will override the [value] property to store the actual platform parameter value.`

As another thought, could this be a generic class rather than an interface?

Reply:
This was point was throughly discussed during proposal reviews and also for the submission of additional info. This approach has been suggested by @BenHenning and @vinitamurthi collectively and it matches the way in which we implement a testable feature. (Help us to provide mocks easily). So @vinitamurthi is the best person to take a look at your suggestion.

Reply:
Sounds good, I must've missed that part. Thanks for the clarification!

Reply:
We will discuss on this in the team meet as there was a email thread regarding the discussion

Reply:
It looks like we're sticking with the interface implementation -- could you highlight the main reasons for this decision? I know we discussed it but could be useful as a record for others viewing this PR.

Reply:
We are sticking with the Interface impl of PlatformParamter so as to be consistent in the way we test features in our codebase. After discussion there was specific point on why we cannot use a Generic Class instead of Generic Interface.

Reply:
What was the specific point? I remember there was a chain of errors occurring when I played around with it but it might be useful to document those here as well if you remember!

Reply:
Sorry for the mistake. I actually meant that there was "no specific point on why we cannot use a Generic Class"